File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    
    for c in s:
        stack.append(c)
        if len(stack) > 1 and stack[-1] == stack[-2]:
            stack.pop()
            stack.pop()
    
    if stack:
        return 0
    else:
        return 1
    

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.swift
import Foundation

func solution(_ s:String) -> Int{
    var answer = 0
    var stack = [Character]()
    
    for c in s {
        if !stack.isEmpty && stack[stack.count - 1] == c {
            stack.removeLast()
        } else {
            stack.append(c)
        }
    }

    return stack.isEmpty ? 1 : 0
}